フォークしたリポジトリからのPull RequestでSonarCloudの実行が失敗する

## その機能リクエストは何らかの問題に関連しますか / Is your feature request related to a problem?

https://github.com/cocoa-mhlw/cocoa/pull/996

https://github.com/cocoa-mhlw/cocoa/runs/6409746066?check_suite_focus=true

こんな感じで、フォークしたリポジトリからのPRについてはSonarCloudの実行が必ず失敗する。

```
Run ./.sonar/scanner/dotnet-sonarscanner begin /k:"cocoa-mhlw_cocoa-app" /o:"cocoa-mhlw" /d:sonar.login="" /d:sonar.host.url="https://sonarcloud.io/" /d:sonar.cs.opencover.reportsPaths="**/coverage.opencover.xml"
SonarScanner for MSBuild 5.5.3
Using the .NET Core version of the Scanner for MSBuild
The format of the analysis property sonar.login= is invalid
Default properties file was found at /Users/runner/work/cocoa/cocoa/.sonar/scanner/.store/dotnet-sonarscanner/5.5.3/dotnet-sonarscanner/5.5.3/tools/net5.0/any/SonarQube.Analysis.xml
Loading analysis properties from /Users/runner/work/cocoa/cocoa/.sonar/scanner/.store/dotnet-sonarscanner/5.5.3/dotnet-sonarscanner/5.5.3/tools/net5.0/any/SonarQube.Analysis.xml
Error: Process completed with exit code 1.
```

`/d:sonar.login`が空文字になっている。

`/d:sonar.login`に`/d:sonar.login="${{ secrets.SONAR_TOKEN }}"`が設定されている。このsecretsはリポジトリに設定されていて、フォークしたリポジトリからのPull Requestの時には適用されない。

https://docs.github.com/ja/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

> 注釈: GITHUB_TOKENを除き、フォークしたリポジトリからワークフローがトリガーされた場合、シークレットはランナーに渡されません。

## 解決策についてお書きください / Describe the solution you'd like

フォークしたリポジトリからのPull Requestでも正常にSonarCloudが実行される方法を検討する。

## あなたが考える代替案についてご説明ください / Describe alternatives you've considered

SonarCloudに失敗してもGitHub Actionsを失敗扱いにしない。

## その他 / Additional context

まずは失敗扱いにしないようにして、その後、SonarCloudが正常実行される方策を考える。

--- a/.github/workflows/CodeAnalysis-Mobile.yml
+++ b/.github/workflows/CodeAnalysis-Mobile.yml
@@ -61,7 +61,8 @@ jobs:
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
     - name: Check SonarScanner Error
       if: ${{ steps.begin_sonarscanner.outcome == 'failure' }}
-      run: exit 1
+      run: |
+        exit 0
     - name: Setup Xamarin
       uses: maxim-lobanov/setup-xamarin@v1
       with:
@@ -81,6 +82,7 @@ jobs:
       env:
         DOTNET_CLI_TELEMETRY_OPTOUT: true
     - name: End SonarScanner
+      if: ${{ steps.begin_sonarscanner.outcome == 'success' }}
       run: ./.sonar/scanner/dotnet-sonarscanner end /d:sonar.login="${{ secrets.SONAR_TOKEN }}"
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/CodeAnalysis-Mobile.yml
+++ b/.github/workflows/CodeAnalysis-Mobile.yml
@@ -53,10 +53,15 @@ jobs:
         New-Item -Path ./.sonar/scanner -ItemType Directory
         dotnet tool update dotnet-sonarscanner --tool-path ./.sonar/scanner
     - name: Begin SonarScanner
+      id: begin_sonarscanner
+      continue-on-error: true
       run: ./.sonar/scanner/dotnet-sonarscanner begin /k:"cocoa-mhlw_cocoa-app" /o:"cocoa-mhlw" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.opencover.reportsPaths="**/coverage.opencover.xml"
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+    - name: Check SonarScanner Error
+      if: ${{ steps.begin_sonarscanner.outcome == 'failure' }}
+      run: exit 1
     - name: Setup Xamarin
       uses: maxim-lobanov/setup-xamarin@v1
       with:

--- a/.github/workflows/CodeAnalysis-Mobile.yml
+++ b/.github/workflows/CodeAnalysis-Mobile.yml
@@ -59,10 +59,6 @@ jobs:
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-    - name: Check SonarScanner Error
-      if: ${{ steps.begin_sonarscanner.outcome == 'failure' }}
-      run: |
-        exit 0
     - name: Setup Xamarin
       uses: maxim-lobanov/setup-xamarin@v1
       with:

